src/hotspot/share/gc/g1/heapRegionSet.cpp

```diff
@@ -276,6 +278,17 @@ void FreeRegionList::remove_starting_at(HeapRegion* first, uint num_regions) {
     curr = next;
   }
 
+  if (prev == NULL) {
```

Comment:
A brief doc on what `prev` and `next` point to could be nice; sth like: `prev` points to the node right before `first` or null when `first == _head`, etc.

Reply:
Thanks Albert for pointing out these, I have changed as you suggested.